app/services/feature_flag.rb

```diff
@@ -65,12 +68,23 @@ def self.deactivate(feature_name)
   def self.active?(feature_name)
     raise unless feature_name.in?(FEATURES)
 
-    FEATURES[feature_name].feature.active?
+    feature_statuses[feature_name].presence || false
   end
 
   def self.sync_with_database(feature_name, active)
     feature = Feature.find_or_initialize_by(name: feature_name)
     feature.active = active
+
     feature.save!
   end
+
+  def self.feature_statuses
+    Rails.cache.fetch(cache_key, expires_in: CACHE_EXPIRES_IN) do
+      Feature.where(name: FEATURES.keys).pluck(:name, :active).to_h.with_indifferent_access
+    end
+  end
+
+  def self.cache_key
+    CacheKey.generate(FEATURE_FLAG_STATUSES_CACHE_KEY)
```

Comment:
this would invalidate cache between deploys (SHA in the cache key),
since the CacheKey already checks for FeatureFlag updated at, wonder if we can effectively set a longer expiration time like a week and also introduce a method like `CacheKey.generate_without_sha()`

Reply:
Good question, so I think we do need the SHA in this instance because the `FEATURES` constant might change between deploys, as in we might add a new feature which has not been persisted yet. The SHA ensures we invalidate the cache and show that